jobsdb/backup.go

```diff
@@ -72,37 +72,40 @@ func (jd *HandleT) backupDSLoop(ctx context.Context) {
 			opPayload, err := json.Marshal(&backupDS)
 			jd.assertError(err)
 
-			opID, err := jd.JournalMarkStart(backupDSOperation, opPayload)
-			if err != nil {
-				return fmt.Errorf("mark start of backup operation: %w", err)
-			}
-			err = jd.backupDS(ctx, backupDSRange)
-			if err != nil {
-				return fmt.Errorf("backup dataset: %w", err)
-			}
-			err = jd.JournalMarkDone(opID)
-			if err != nil {
-				return fmt.Errorf("mark end of backup operation: %w", err)
+			if err := jd.WithTx(func(tx *Tx) error {
+				opID, err := jd.JournalMarkStartInTx(tx, backupDSOperation, opPayload)
+				if err != nil {
+					return fmt.Errorf("mark start of backup operation: %w", err)
+				}
+				if err := jd.backupDS(ctx, backupDSRange); err != nil {
+					return fmt.Errorf("backup dataset: %w", err)
+				}
+				if err := jd.journalMarkDoneInTx(tx, opID); err != nil {
+					return fmt.Errorf("mark end of backup operation: %w", err)
+				}
+				return nil
+			}); err != nil {
```

Comment:
One of the usages of the journal is detecting if a job got interrupted, so it could recover a half-completed job later.

With this approach here, since `backupDS` is not using the transaction we effectively add a journal entry once the backup job is completed. I am not sure how this will impact our recovery logic, and if it is useful to use the journal pattern like this.

Reply:
It looks like we are using journal to detect incomplete backups and run the following code:

```go
func (jd *HandleT) removeTableJSONDumps() {
	backupPathDirName := "/rudder-s3-dumps/"
	tmpDirPath, err := misc.CreateTMPDIR()
	jd.assertError(err)
	files, err := filepath.Glob(fmt.Sprintf("%v%v_job*", tmpDirPath+backupPathDirName, jd.tablePrefix))
	jd.assertError(err)
	for _, f := range files {
		err = os.Remove(f)
		jd.assertError(err)
	}
}
``` 

The cleanup is trivial and we can probably run it after every start, regardless of a successful backup or not. The os tmp folder can also take care of this cleanup doing a system restart.

I would propose to completely remove journaling for backups, we will get right of a lot of code/complexity.

Reply:
NOTE: We need to think a bit more about `backupDropDSOperation`,  the next section.

Reply:
Currently journaling is mostly used for logging, that's the reason we haven't removed it yet. Journal table can provide useful information for debugging jobsdb issues.

Reply:
@atzoum @cisse21 Please let me know your thoughts on this as well

Reply:
Still wrapping with transactions doesn't help. 
* For debugging you will never know if something was started and later interrupted. 
* The recovery logic gets disabled with this change, so by keeping the code there we are just adding more confusion into the mix

Reply:
We can remove recovery logic altogether, we can also do this as part of a another PR.

Reply:
Sounds fair.

Does it make sense to use journaling mart start/done inside the same transaction?

Note that during recovery, entries can be deleted:
```go
	if undoOp {
		sqlStatement = fmt.Sprintf(`DELETE from "%s_journal" WHERE id=$1`, jd.tablePrefix)
```

Do you have any concerns about using logging instead of a journal? Is it a reliability concern?